### DISCUSSION TITLE
Rollever for teaching R coding and for defining words in general

### DISCUSSION THREAD

**Discussion by harrelfe** [quarto-dev/quarto-cli#1428] [Q&A] [answered]
This regular expression learning tool is excellent: https://regexr.com

It would be great if `Quarto` had a way to allow us to explain piece by piece the elements of a sentence or of a complex R expression, just as rolling the mouse over elements of the regexp causes explanatory boxes to pop up.

**Comment by jjallaire** [comment, marked as answer]
Yes, I use regex all the time to write my own regexes! Agree that this would be an incredible thing to do for R code. Note that its possible that this could be implemented as an extension: https://quarto.org/docs/extensions/
